docs/guides/tutorials/trpc.rst

```diff
@@ -0,0 +1,468 @@
+and context in both **Next.js** and **Express** environments. This will ensure 
+that user authentication is handled securely and that both server-side and 
+client-side tRPC calls can access the user’s session.
+
+EdgeDB Auth with tRPC and tRPC context in Next.js
+-------------------------------------------------
+
+In **Next.js**, integrating **EdgeDB Auth** with **tRPC** involves creating a 
+context that provides the user session and EdgeDB client to the tRPC API.
+
+1. **Initialize EdgeDB Client and Auth**
+
+   First, initialize the **EdgeDB** client and **EdgeDB Auth**:
+
+   .. code-block:: typescript
+
+      import { createClient } from "edgedb";
+      import createAuth from "@edgedb/auth-nextjs/app";
+
+      // Initialize EdgeDB client
+      export const edgedbClient = createClient();
+
+      // Initialize EdgeDB Auth
+      export const auth = createAuth(edgedbClient, {
+        baseUrl: process.env.VERCEL_ENV === "production"
+          ? "https://production.yourapp.com"
+          : "http://localhost:3000",
+      });
+
+2. **Create tRPC Context**
+
+   The **tRPC** context provides the session and the EdgeDB client to the tRPC 
+   procedures:
+
+   .. code-block:: typescript
+
+      import { initTRPC } from '@trpc/server';
+      import { headers } from "next/headers";
+      import { auth } from "./path-to-auth";
+      import { edgedbClient } from "./path-to-edgedb";
+
+      // Create tRPC context with session and EdgeDB client
+      export const createTRPCContext = async () => {
+        const session = await auth.getSession(); // Retrieve session from EdgeDB Auth
+
+        return {
+          session, // Pass the session to the context
+          edgedbClient, // EdgeDB client for querying the database
+        };
+      };
+
+      // Initialize tRPC with context
+      const t = initTRPC.context<typeof createTRPCContext>().create({});
+
+3. **Use tRPC Context in API Handler**
+
+   In **Next.js**, set up an API handler to connect your **tRPC router** with 
+   the context:
+
+   .. code-block:: typescript
+
+      import { createNextApiHandler } from '@trpc/server/adapters/next';
+      import { appRouter } from './path-to-router';
+
+      export default createNextApiHandler({
+        router: appRouter, // Your tRPC router
+        createContext: createTRPCContext,
+      });
+
+4. **Example tRPC Procedure**
+
+   You can now write procedures in your tRPC router, making use of the 
+   **EdgeDB Auth** session and the **EdgeDB** client:
+
+   .. code-block:: typescript
+
+      export const appRouter = t.router({
+        getUserData: t.procedure.query(async ({ ctx }) => {
+          if (!ctx.session) {
+            throw new Error("Not authenticated");
+          }
+          // Fetch data from EdgeDB using the client's session
+          const userData = await ctx.edgedbClient.query(`
+            SELECT User { name, email } WHERE .id = <uuid>$id
+          `, { id: ctx.session.user.id });
```

Comment:
```suggestion
          if (!(await ctx.session.isSignedIn())) {
            throw new Error("Not authenticated");
          }
          // Fetch data from EdgeDB using the authenticated client
          const userData = await ctx.session.client.query(`
            select User { name, email }
          `);
```

This assumes you've set up some access policies to disallow selecting users other than yourself. Otherwise:

```suggestion
          if (!(await ctx.session.isSignedIn())) {
            throw new Error("Not authenticated");
          }
          // Fetch data from EdgeDB using the authenticated client
          const userData = await ctx.session.client.query(`
            select User { name, email } filter global ext::auth::ClientTokenIdentity in .identities
          `);
```

(or for single identity<->user models `global ext::auth::ClientTokenIdentity = .identity`)